framework/src/components/storage/entities/account.js

```diff
@@ -677,6 +723,85 @@ class Account extends BaseEntity {
 		);
 	}
 
+	/**
+	 * Update dependent records for the account
+	 * Delete dependent record for the account
+	 *
+	 * @param {string} dependencyName - Name of the dependent table
+	 * @param {string} address - Address of the account
+	 * @param {string} dependentPublicKey - Dependent public id
+	 * @param {Object} [tx] - Transaction object
+	 * @return {*}
+	 */
+	updateDependentRecords(dependencyName, address, dependentPublicKeys, tx) {
+		assert(
+			Object.keys(dependentFieldsTableMap).includes(dependencyName),
+			`Invalid dependency name "${dependencyName}" provided.`
+		);
+
+		const sqlForDelete = this.SQLs.deleteDependentRecords;
+		const sqlForInsert = this.SQLs.createDependentRecords;
+		const tableName = dependentFieldsTableMap[dependencyName];
+
+		return this.adapter
+			.execute(
+				`SELECT "dependentId" FROM ${tableName} WHERE "accountId" = $1`,
+				[address]
+			)
+			.then(dependentRecordsForAddress => {
```

Comment:
The same here. This then looks not so syntactic sugar. With await it would be much better.

Reply:
I agree with Diego here. It is not very readable as the `return` is on the 9th line but the code goes on a whole lot further. With `await` you can split the code further up in an extra function which you can await to reduce the size of the initial function.

Separate function for: `return Promise.all(dependentRecordPromises);`

Reply:
@diego-G converted. 
@michielmulders code changed. I could probably agree with more refactors but they're outside the scope of this PR which we need merged ASAP. If you feel more refactors could apply please feel free to report an issue for approaching it for all entities.